cpp/src/copying/scatter.cu

```diff
@@ -0,0 +1,376 @@
+    gather_bitmask(source, gather_map.begin(), result, stream);
+
+    return std::make_unique<table>(std::move(result));
+  }
+
+  template <typename T, std::enable_if_t<not std::is_integral<T>::value
+      or std::is_same<T, bool8>::value>* = nullptr>
+  std::unique_ptr<table> operator()(
+      table_view const& source, column_view const& scatter_map,
+      table_view const& target, bool check_bounds,
+      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    CUDF_FAIL("Scatter map column must be an integral, non-boolean type");
+  }
+};
+
+template <bool mark_true, typename MapIterator>
+__global__ void marking_bitmask_kernel(
+    mutable_column_device_view destination,
+    MapIterator scatter_map,
+    size_type num_scatter_rows)
+{
+  size_type row = threadIdx.x + blockIdx.x * blockDim.x;
+
+  while (row < num_scatter_rows) {
+    size_type const output_row = scatter_map[row];
+
+    if (mark_true){
+      destination.set_valid(output_row);
+    } else {
+      destination.set_null(output_row);
+    }
+
+    row += blockDim.x * gridDim.x;
+  }
+}
+
+template <typename MapIterator>
+void scatter_scalar_bitmask(std::vector<std::unique_ptr<scalar>> const& source,
+    MapIterator scatter_map, size_type num_scatter_rows,
+    std::vector<std::unique_ptr<column>>& target, cudaStream_t stream)
+{
+  constexpr size_type block_size = 256;
+  size_type const grid_size = grid_1d(num_scatter_rows, block_size).num_blocks;
+
+  for (size_t i = 0; i < target.size(); ++i) {
+    if (target[i]->nullable()) {
+      auto target_view = mutable_column_device_view::create(
+        target[i]->mutable_view(), stream);
+
+      auto bitmask_kernel = source[i]->is_valid(stream)
+        ? marking_bitmask_kernel<true, decltype(scatter_map)>
+        : marking_bitmask_kernel<false, decltype(scatter_map)>;
+      bitmask_kernel<<<grid_size, block_size, 0, stream>>>(
+        *target_view, scatter_map, num_scatter_rows);
+    }
+  }
+}
+
+template <typename index_type>
+struct column_scalar_scatterer {
+  template <typename T, std::enable_if_t<is_fixed_width<T>()>* = nullptr>
+  std::unique_ptr<column> operator()(std::unique_ptr<scalar> const& source,
+      column_view const& indices, column_view const& target,
+      rmm::mr::device_memory_resource* mr, cudaStream_t stream)
+  {
+    using cudf::detail::fixed_width_scalar;
+
+    // Result column must have a null mask if the source is not valid
+    auto result = copy_with_nullable(target, not source->is_valid(), mr, stream);
+    auto result_view = result->mutable_view();
+
+    // Transform negative indices to index + target size
+    auto scatter_iter = thrust::make_transform_iterator(
+      indices.begin<index_type>(),
+      index_converter<index_type>{target.size()});
+
+    // Use permutation iterator with constant index to dereference scalar data
+    auto scalar_impl = static_cast<fixed_width_scalar<T>*>(source.get());
```

Comment:
You can use scalar_type_t<T> instead of fixed_width_scalar<T>

Reply:
Ok, will do